TUTORIAL.md

```diff
@@ -1091,6 +1091,94 @@ of the type annotation, or even the function's implementation! The only way to h
 Similarly, the only way to have a function whose type is `a -> a` is if the function's implementation returns
 its argument without modifying it in any way. This is known as [the identity function](https://en.wikipedia.org/wiki/Identity_function).
 
+## Tag Unions
+
+We can also annotate types that include tags:
+
+```coffee
+colorFromStr : Str -> [Red, Green, Yellow]
+colorFromStr = \string ->
+    when string is
+        "red" -> Red
+        "green" -> Green
+        _ -> Yellow
+```
+
+You can read the type `[Red, Green, Yellow]` as "a *tag union* of the tags `Red`, `Green`, and `Yellow`."
+
+Tags are always annotated as being part of a tag union, even if that union only has one tag in it. For example:
+
+```coffee
+redTag : [Red]
+redTag = Red
+```
+
+Tag unions can accumulate additional tags based on how they're used in the program. Consider this `if` expression:
+
+```elm
+\str ->
+    if Str.isEmpty str then
+        "it was empty"
+    else
+        ["it was not empty"]
+```
+
+This will be a type mismatch because the two branches have incompatible types: the first branch
+evaluates to a `Str`, but the `else` branch evaluates to a `List Str`!
```

Comment:
Maybe worth moving to this after the correct (Result) example, or further describing why tags are useful in that they avoid mismatches here? To me it's not immediately clear why this example is important to understanding tag accumulation.

Reply:
I think I can actually just drop this example.